File: src/main/scala/com/github/neo4j/node/Neo4jNode.scala
package com.github.neo4j.node

import org.neo4j.ogm.annotation.{GeneratedValue, Id, Property}

trait Neo4jNode:
  @Id
  protected var id: Long = _

  def getId: Long = id

  def setId(id: Long): Unit = this.id = id

  @Property(name = "creationTimeStamp")
  private var creationTimeStamp: String = _

  def getCreationTimeStamp: String = creationTimeStamp

  def setCreationTimeStamp(creationTimeStamp: String): Unit = this.creationTimeStamp = creationTimeStamp

  @Property(name = "lastUpdateTimeStamp")
  private var lastUpdateTimeStamp: String = _

  def getLastUpdateTimeStamp: String = lastUpdateTimeStamp

  def setLastUpdateTimeStamp(lastUpdateTimeStamp: String): Unit = this.lastUpdateTimeStamp = lastUpdateTimeStamp

end Neo4jNode

File: crud/src/main/java/io/github/abhishekghoshh/crud/neo4j/node/Neo4jNode.java
package io.github.abhishekghoshh.crud.neo4j.node;

import lombok.*;
import org.neo4j.ogm.annotation.GeneratedValue;
import org.neo4j.ogm.annotation.Id;
import org.neo4j.ogm.annotation.Property;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public abstract class Neo4jNode {

    @Id
    @GeneratedValue
    protected Long id;

    @Property(name = "creationTimeStamp")
    private String creationTimeStamp;

    @Property(name = "lastUpdateTimeStamp")
    private String lastUpdateTimeStamp;

}
